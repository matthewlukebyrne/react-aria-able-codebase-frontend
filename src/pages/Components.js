// Images Area
import landing_image from '../img/landing_image.svg';

// Styled Components
import styled from 'styled-components';
import { Link } from 'react-router-dom';


const Components = () => {
  return (
    <StyledWork>
      <StyledPonent>
        <h2>Ponent 1</h2>
        <div className="line"></div>
        <Link>
            <img src={landing_image} alt="ponent_1" />
        </Link>
      </StyledPonent>
      <StyledPonent>
        <h2>Ponent 2</h2>
        <div className="line"></div>
        <Link>
            <img src={landing_image} alt="ponent_2" />
        </Link>
      </StyledPonent>
      <StyledPonent>
        <h2>Ponent 3</h2>
        <div className="line"></div>
        <Link>
            <img src={landing_image} alt="ponent_3" />
        </Link>
      </StyledPonent>
    </StyledWork>
  );
};

export default Components;


// Styled Components
const StyledWork = styled.div`
  min-height: 100vh;
  overflow: hidden;
  padding: 5rem 10rem;
  h2 {
    padding: 1rem 0rem;

  }
`;

const StyledPonent = styled.div`
  padding-bottom: 10rem;
  .line {
    height: 0.5rem;
    background: #e3e3e3;
    margin-bottom: 3rem;
  }

  img {
    width: 100%;
    height: 70vh;
    object-fit: cover;
  }
`;