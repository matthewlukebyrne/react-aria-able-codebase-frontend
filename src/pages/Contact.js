import styled from 'styled-components';
import profile_pic from '../img/profile.jpg';


// Import Framer Motion for Page animations
import { motion } from 'framer-motion';
import { pageAnimation } from '../animation';

const Contact = () => {
  return (
    <StyledContact
      exit='exit'
      variants={pageAnimation}
      initial='hidden'
      animate='show'
    >
      <div>
        <StyledSocial>
          <StyledCircle />
          <a
            href='https://www.linkedin.com/in/matthewlukebyrne/'
            target='_blank'
            rel='noreferrer'
          >
            {' '}
            <h2>LinkedIn</h2>
          </a>
        </StyledSocial>
        <StyledSocial>
          <StyledCircle />
          <a
            href='https://github.com/matthewlukebyrne'
            target='_blank'
            rel='noreferrer'
          >
            {' '}
            <h2>GitHub</h2>
          </a>
        </StyledSocial>
        <StyledSocial>
          <StyledCircle />
          <a href='https://www.kooba.ie/' target='_blank' rel='noreferrer'>
            {' '}
            <h2>Kooba</h2>
          </a>
        </StyledSocial>
      </div>
      <StyledContactImage>
        <img src={profile_pic} alt='profile_img' />
      </StyledContactImage>
    </StyledContact>
  );
};

export default Contact;

const StyledContact = styled(motion.div)`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 5rem 10rem;
  color: #e3e3e3;
  min-height: 90vh;
`;

// const StyledTitle = styled.div`
//   margin-bottom: 4rem;
//   color: #1b1b1b;
// `;

// const StyledHide = styled.div`
//   overflow: hidden;
// `;

const StyledSocial = styled.div`
  display: flex;
  align-items: center;
  h2 {
    margin: 2rem;
  }

  @media (max-width: 1300px) {
    display: flex;
    flex-direction: column;
  }
`;

const StyledCircle = styled.div`
  border-radius: 50%;
  width: 3rem;
  height: 3rem;
  background-color: #f5f5f5;
`;

const StyledContactImage = styled.div`
  img {
    width: 100%;
    height: 50vh;
    object-fit: cover;
    border-radius: 50%;
  }

  /* Added Media Queries */
  @media (max-width: 1300px) {
    img {
      display: none;
    }
  }
`;
