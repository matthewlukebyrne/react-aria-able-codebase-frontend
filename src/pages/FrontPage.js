import FAQSection from '../components/FAQSection';
import FeatureSection from '../components/FeatureSection';
import LandingSection from '../components/LandingSection';

// Import Framer Motion for Page animations
import { motion } from 'framer-motion';
import { pageAnimation } from '../animation';

const FrontPage = () => {
  return (
    <div data-testid='frontpagetest-id'>
      <motion.div
        exit='exit'
        variants={pageAnimation}
        initial='hidden'
        animate='show'
      >
        <LandingSection />
        <FeatureSection />
        <FAQSection />
      </motion.div>
    </div>
  );
};

export default FrontPage;
