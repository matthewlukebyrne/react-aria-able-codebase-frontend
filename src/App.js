// Global Styles
import GlobalStyle from './components/GlobalStyle';
// Import Elements
import FrontPage from './pages/FrontPage';
// Import the Navbar
import Navbar from './components/Navbar';
// Import the Contact Page
import Contact from './pages/Contact';
// Import the Contact Page
import Components from './pages/Components';
/*
  Using React Router we can direct the pages we want to go to
  The Route Facilates the full routing of the application
  A Switch stops when it comes across the EXACT URL
*/
import {BrowserRouter, Switch, Route } from 'react-router-dom';
import NotFound from './pages/NotFound';

const App = () => {
  return (
    <div data-testid='apptest-id' className='App'>
      <BrowserRouter>
        <GlobalStyle />
        <Navbar />
        <Switch>
          {/* Home Path */}
          <Route path='/' exact>
            <FrontPage />
          </Route>
          {/* Components Path */}
          <Route path='/components'>
            <Components />
          </Route>
          {/* Contact Path */}
          <Route path='/contact'>
            <Contact />
          </Route>
          {/* If all else fails 404 not Found */}
          <Route>
            <NotFound />
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
};

export default App;
