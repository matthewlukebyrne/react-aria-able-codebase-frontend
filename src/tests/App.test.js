// Imports from a combination of Jest, React Library and Enzyme
import { render, screen } from '@testing-library/react';
import { shallow, mount } from 'enzyme';
import App from '../App';
import Components from '../pages/Components';
import Contact from '../pages/Contact';
import FrontPage from '../pages/FrontPage';
import { MemoryRouter } from 'react-router-dom';
import { NotFound } from 'http-errors';

// Full description of the test suite (App Component)
// TDD - Red Green Refactor
describe('App Components Unit Tests', () => {
  it('render the primary component without crashing', () => {
    render(<App />);
    const app = screen.getByTestId('apptest-id');
    // make assertions
    expect(app).toBeInTheDocument();
  });

  it('render the Front Page and only expects one render from App Component', () => {
    const appWrapper = shallow(<App />);
    const frontPage = appWrapper.find(FrontPage);
    // assertion
    expect(frontPage).toHaveLength(1);
  });

  it('render the Components Page and only expects one render from App Component', () => {
    const appWrapper = shallow(<App />);
    const frontPage = appWrapper.find(Components);
    // assertion
    expect(frontPage).toHaveLength(1);
  });

  it('render the Contact Page and only expects one render from App Component', () => {
    const appWrapper = shallow(<App />);
    const frontPage = appWrapper.find(Contact);
    // assertion
    expect(frontPage).toHaveLength(1);
  });

  // Testing
  it('invalid path should redirect to 404 Not Found Page', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/test']}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(FrontPage)).toHaveLength(1);
    expect(wrapper.find(NotFound)).toHaveLength(0);
  });
});
