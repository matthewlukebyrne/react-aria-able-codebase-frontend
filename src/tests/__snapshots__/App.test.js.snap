// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components Unit Tests should take a snapshot of the entire Application 1`] = `
<DocumentFragment>
  .c1 {
  min-height: 90vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0rem 10rem;
  color: white;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-right: 5rem;
  z-index: 2;
}

.c2 h2 {
  padding-bottom: 1rem;
  font-weight: lighter;
}

.c4 {
  z-index: 2;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
}

.c4 img {
  padding: 5rem;
  width: 100%;
  height: 100vh;
}

.c3 {
  overflow: hidden;
}

.c6 {
  display: block;
}

.c6 span {
  display: block;
}

.c6 h2 {
  padding-top: 5rem;
  font-weight: lighter;
}

.c6 .faq-line {
  background: #cccccc;
  height: 0.2rem;
  margin: 2rem 0rem;
  width: 100%;
}

.c6 .question {
  padding: 3rem 0rem;
  cursor: pointer;
}

.c6 .answer {
  padding: 2rem 0rem;
}

.c6 .answer p {
  padding: 1rem 0rem;
}

.c5 {
  background-color: whitesmoke;
  color: #333;
}

.c5 h2,
.c5 p {
  padding: 3rem 0rem;
  color: #333;
}

.c0 {
  min-height: 10vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0rem 5rem;
  background-color: whitesmoke;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 10;
}

.c0 a {
  color: #1b1b1b;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 3rem;
}

.c0 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
}

.c0 li {
  padding-left: 5rem;
  position: relative;
}

@media (max-width:1300px) {

}

@media (max-width:1300px) {
  .c1 {
    display: block;
    padding: 5rem 2rem;
    text-align: center;
  }
}

@media (max-width:1300px) {
  .c2 {
    display: block;
    padding: 5rem 2rem;
    text-align: center;
  }
}

@media (max-width:1300px) {
  .c4 {
    display: block;
    padding: 5rem 2rem;
    text-align: center;
  }

  .c4 img {
    width: 100%;
    height: 50vh;
    object-fit: cover;
  }
}

@media (max-width:1300px) {
  .c6 {
    display: block;
    padding: 5rem 2rem;
    text-align: center;
  }
}

@media (max-width:1300px) {
  .c5 {
    display: none;
  }
}

@media (max-width:1300px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding: 2rem;
  }

  .c0 #main_logo {
    display: inline-block;
    margin: 1rem;
  }

  .c0 ul {
    padding: 2rem 1rem;
    -webkit-box-pack: space-evenly;
    -webkit-justify-content: space-evenly;
    -ms-flex-pack: space-evenly;
    justify-content: space-evenly;
    width: 100%;
  }

  .c0 ul li {
    padding: 0;
  }
}

<div
    class="App"
    data-testid="apptest-id"
  >
    <nav
      class="c0"
      data-testid="navtest-id"
    >
      <h1>
        <a
          href="/"
        >
          Aria-Able
        </a>
      </h1>
      <ul>
        <li>
          <a
            href="/components"
          >
            Components
          </a>
        </li>
        <li>
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <div
      data-testid="frontpagetest-id"
    >
      <div
        class="c1"
        data-testid="landingtest-id"
      >
        <div
          class="c2"
        >
          <div
            class="title"
          >
            <div
              class="c3"
            >
              <h2>
                Accessible
              </h2>
            </div>
            <div
              class="c3"
            >
              <h2>
                 styled 
                <span>
                   components 
                </span>
              </h2>
            </div>
            <div
              class="c3"
            >
              <h2>
                 all in one place...
              </h2>
            </div>
          </div>
          <p>
            Check out the project today!
          </p>
          <button>
            Find Out More
          </button>
        </div>
        <div
          class="c4"
        >
          <img
            alt="landing_image"
            src="landing_image.svg"
          />
        </div>
      </div>
      <div
        class="c1 c5"
        data-testid="featuretest-id"
      >
        <div
          class="c2"
        >
          <h2>
            People should 
            <span>
              perceive
            </span>
             understand, navigate, and 
            <span>
              interact
            </span>
             with the web.
          </h2>
          <div
            class="cards"
          >
            <div
              class="card"
            >
              <div
                class="icon"
              >
                <svg
                  aria-labelledby="svg-inline--fa-title-CByPc2cbAwJr"
                  class="svg-inline--fa fa-universal-access fa-w-16 "
                  data-icon="universal-access"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="svg-inline--fa-title-CByPc2cbAwJr"
                  >
                    icon-1
                  </title>
                  <path
                    d="M256 48c114.953 0 208 93.029 208 208 0 114.953-93.029 208-208 208-114.953 0-208-93.029-208-208 0-114.953 93.029-208 208-208m0-40C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 56C149.961 64 64 149.961 64 256s85.961 192 192 192 192-85.961 192-192S362.039 64 256 64zm0 44c19.882 0 36 16.118 36 36s-16.118 36-36 36-36-16.118-36-36 16.118-36 36-36zm117.741 98.023c-28.712 6.779-55.511 12.748-82.14 15.807.851 101.023 12.306 123.052 25.037 155.621 3.617 9.26-.957 19.698-10.217 23.315-9.261 3.617-19.699-.957-23.316-10.217-8.705-22.308-17.086-40.636-22.261-78.549h-9.686c-5.167 37.851-13.534 56.208-22.262 78.549-3.615 9.255-14.05 13.836-23.315 10.217-9.26-3.617-13.834-14.056-10.217-23.315 12.713-32.541 24.185-54.541 25.037-155.621-26.629-3.058-53.428-9.027-82.141-15.807-8.6-2.031-13.926-10.648-11.895-19.249s10.647-13.926 19.249-11.895c96.686 22.829 124.283 22.783 220.775 0 8.599-2.03 17.218 3.294 19.249 11.895 2.029 8.601-3.297 17.219-11.897 19.249z"
                    fill="currentColor"
                  />
                </svg>
                <h3>
                  A11y Initiative and WCAG guidelines standards.
                </h3>
              </div>
              <p>
                Components that have ARIA roles attached and elements that keep high consistent standards across semantic descriptive HTML5 atrributes.
              </p>
            </div>
            <div
              class="card"
            >
              <div
                class="icon"
              >
                <svg
                  aria-labelledby="svg-inline--fa-title-WzcEjo3YGtvh"
                  class="svg-inline--fa fa-book fa-w-14 "
                  data-icon="book"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="svg-inline--fa-title-WzcEjo3YGtvh"
                  >
                    icon-2
                  </title>
                  <path
                    d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                    fill="currentColor"
                  />
                </svg>
                <h3>
                  Documentation at hand.
                </h3>
              </div>
              <p>
                Everything in one place and easy to find for developers. A documentation tab is provided with code snippets explaining functionality.
              </p>
            </div>
            <div
              class="card"
            >
              <div
                class="icon"
              >
                <svg
                  aria-labelledby="svg-inline--fa-title-LYspesJ9xsZx"
                  class="svg-inline--fa fa-book-reader fa-w-16 "
                  data-icon="book-reader"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="svg-inline--fa-title-LYspesJ9xsZx"
                  >
                    icon-3
                  </title>
                  <path
                    d="M352 96c0-53.02-42.98-96-96-96s-96 42.98-96 96 42.98 96 96 96 96-42.98 96-96zM233.59 241.1c-59.33-36.32-155.43-46.3-203.79-49.05C13.55 191.13 0 203.51 0 219.14v222.8c0 14.33 11.59 26.28 26.49 27.05 43.66 2.29 131.99 10.68 193.04 41.43 9.37 4.72 20.48-1.71 20.48-11.87V252.56c-.01-4.67-2.32-8.95-6.42-11.46zm248.61-49.05c-48.35 2.74-144.46 12.73-203.78 49.05-4.1 2.51-6.41 6.96-6.41 11.63v245.79c0 10.19 11.14 16.63 20.54 11.9 61.04-30.72 149.32-39.11 192.97-41.4 14.9-.78 26.49-12.73 26.49-27.06V219.14c-.01-15.63-13.56-28.01-29.81-27.09z"
                    fill="currentColor"
                  />
                </svg>
                <h3>
                  Technology and Screen Readers Features
                </h3>
              </div>
              <p>
                Screen readers activating ARIA roles insuring the people who can not traditonally access the web are getting the same user experience. This can be showcased in the testing area!
              </p>
            </div>
            <div
              class="card"
            >
              <div
                class="icon"
              >
                <svg
                  aria-labelledby="svg-inline--fa-title-Xi9R1SNVmAQU"
                  class="svg-inline--fa fa-vials fa-w-20 "
                  data-icon="vials"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 640 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="svg-inline--fa-title-Xi9R1SNVmAQU"
                  >
                    icon-4
                  </title>
                  <path
                    d="M72 64h24v240c0 44.1 35.9 80 80 80s80-35.9 80-80V64h24c4.4 0 8-3.6 8-8V8c0-4.4-3.6-8-8-8H72c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm72 0h64v96h-64V64zm480 384H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h608c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zM360 64h24v240c0 44.1 35.9 80 80 80s80-35.9 80-80V64h24c4.4 0 8-3.6 8-8V8c0-4.4-3.6-8-8-8H360c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm72 0h64v96h-64V64z"
                    fill="currentColor"
                  />
                </svg>
                <h3>
                  Developed with React Testing Library, Jest and Enzyme.
                </h3>
              </div>
              <p>
                Custom components that are tested and proccessed to ensure that they are performing to the best of the ability with no side affects given to the user.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c4"
      >
        <img
          alt="features_image"
          src="features_snap.png"
        />
      </div>
      <div
        class="c1 c6"
        data-testid="faqtest-id"
      >
        <h2>
          Questions 
          <span>
            FAQ
          </span>
        </h2>
        <div
          class="question"
        >
          <h4>
            What is Aria-Able? What is the concept all about?
          </h4>
          <div
            class="answer"
          >
            <p>
              A area developed for developers for accessible web component testing.
            </p>
            <p>
              Delivered through Storybook JS a detailed design library for elements; the creation of a accessible library of React components encapsulated with logic that pass guidelines to accessiblity officially with Web Content Accessibility Guidelines (WCAG).
              <br />
              <br />
              Accessible Rich Internet Applications (ARIA) is a set of attributes that define ways to make web content and web applications (especially those developed with JavaScript) more accessible to people with disabilities.
            </p>
          </div>
          <div
            class="faq-line"
          />
        </div>
        <div
          class="question"
        >
          <h4>
            What else can I do with the components to ensure they are fully accessible?
          </h4>
          <div
            class="answer"
          >
            <p>
              Features including Actions, Knobs, A11y.
            </p>
            <p>
              Along with the ability to insert in and out components you can check for color contrast themes, actions, and click events registered through the DOM. This is delivered through Storybook JS addon features entitled Knobs and Actions, and A11y that is integrated into the Storybook framework.
            </p>
          </div>
          <div
            class="faq-line"
          />
        </div>
        <div
          class="question"
        >
          <h4>
            Is there a documentation area?
          </h4>
          <div
            class="answer"
          >
            <p>
              Full descriptions and details.
            </p>
            <p>
              With the documentation provided through official ARIA supported channels such as Mozilla Developer Network (MDN) documentation is provided for each element and role that was inplemented on the Storybook platform. There is a details section stating what exactly each components does and a example section of code to compliment.
            </p>
          </div>
          <div
            class="faq-line"
          />
        </div>
        <div
          class="question"
        >
          <h4>
            Do I have to pay for these? Are these components open source?
          </h4>
          <div
            class="answer"
          >
            <p>
              Free of charge.
            </p>
            <p>
              These components were created by developers and for developers; increasing usability across web applications and the internet. Coming soon is the ability to pull down these components via NPM and Yarn package managers respectively; where you can import and declare in your code the ARIA implementation to any open source projects!
            </p>
          </div>
          <div
            class="faq-line"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
